urlc3ca8aa24b_davidsan_rivercityransom_tgz-pJ8-rivercityransom_dirtytests_GangsterTestMainJ8/src/rivercityransom/dirtytests/BlocTestMain.java

```diff
@@ -0,0 +1,51 @@
+package rivercityransom.dirtytests;
+
+import rivercityransom.contracts.BlocContract;
+import rivercityransom.contracts.ObjetContract;
+import rivercityransom.enumerations.Tresor;
+import rivercityransom.enumerations.Type;
+import rivercityransom.impl.BlocImpl;
+import rivercityransom.impl.ObjetImpl;
+import rivercityransom.services.BlocService;
+
+public class BlocTestMain {
+
+	// Ca teste les BlocService et les ObjetService
+
+	public static void test1() {
+		BlocService b = new BlocContract(new BlocImpl());
+		b.init(Type.FOSSE, null);
+		System.out.println(b);
+	}
+
+	public static void test2() {
+		BlocService b = new BlocContract(new BlocImpl());
+		b.init(Type.VIDE, null);
+		System.out.println(b);
+	}
+
+	public static void test3() {
+		BlocService b = new BlocContract(new BlocImpl());
+		ObjetContract o = new ObjetContract(new ObjetImpl());
+		o.init(Tresor.CHAINEDEVELO);
+		b.init(Type.VIDE, o);
+		System.out.println(b);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 28:32, 36:40.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.